### ISSUE TITLE
How to check the loss function for object detection

### ISSUE THREAD

**Issue by FrancescoSaverioZuppichini** [PyTorchLightning/lightning-flash#828]
Hi guys,

I hope you are doing great. The code for object detection is quite crazy, with a lot of back and forth between different integration. I would like to ask if you can point me out where in the code the loss functions for object detection or the logic is defined. I see that there is a `IceVisionAdapter` in `core.integrations.icevision.adapter` that takes an `icevision_adapter` as a parameter, so I am not sure if that is the actual adapter or just a decorator. Any help would be nice.

Thank you!

Francesco

**Comment by ethanwharris** [comment]
Hi @FrancescoSaverioZuppichini, thanks for the question. Currently, all of our object detection models are provided by icevision. Tha adapter works by allowing the icevision component to compute the loss internally. The reason for this is that each of the frameworks that icevision uses (mmdet, efficientdet, torchvision, etc.) bundle the loss function internally as part of the model. So who owns the loss function is different depending on which framework the head / backbone is coming from. I think that some of those frameworks may have some configuration options (e.g. this may be possible with mmdet) but I'm not sure. It would be worth looking in to whether this is possible (and how to do it) in icevision as we are only a thin wrapper so anything that works there should also work in Flash :smiley:

**Comment by FrancescoSaverioZuppichini** [comment]
cool, i'll check with them! I didn't know about icevision, really cool library. Thank for answering ;)
